packages/scaffold-ui/src/partials/w3m-connecting-wc-browser/index.ts

```diff
@@ -29,15 +29,20 @@ export class W3mConnectingWcBrowser extends W3mConnectingWidget {
     try {
       this.error = false
       const { connectors } = ConnectorController.state
-      const announcedConnector = connectors.find(
-        c => c.type === 'ANNOUNCED' && c.info?.rdns === this.wallet?.rdns
+
+      const connector = connectors.find(
+        c =>
+          (c.type === 'ANNOUNCED' && c.info?.rdns === this.wallet?.rdns) ||
+          c.type === 'INJECTED' ||
+          c.name === this.wallet?.name
```

Comment:
should you match by ID instead?

Reply:
I think this way makes more explicit that we are matching the name. In case of Solana wallets we would need this to be `c.id === this.wallet?.name` because the id in Solana refers to the wallet name as the Wallet Standard doesn't support ids and the extension do not expose a id.